section-engineering-education/engineering-education
# Improve build time

**Description**:
Improve the build time of the EngEd repo.
- Converted the following partials to partialCached components reducing the build time and number of times each partial is processed/computed:
  - `footer.html`
  - `header.html`
  - `eng-ed-banner.html`
  - `footer-scripts.html`
  - `head-scripts.html`
- Added `test-build.yml` GitHub action to test the build on a PR to master.

**How to Test**:
Compare results of running `hugo --templateMetrics --templateMetricsHints` on both master and this branch locally. Build time results will vary with each build. However the results should show a decrease in build time as well as a decrease in the `count` column for many of the partials with a high `cache potential` score.

Initial build time report:
![image](https://user-images.githubusercontent.com/15935329/164281435-887b74d4-9857-4d76-a421-091a4b9e8ca8.png)

Improved build time report. reduced build by about 1 second:
![image](https://user-images.githubusercontent.com/15935329/164281514-5c096d78-b426-47b2-a21d-0e573e307382.png)
